app/models/User.js

```diff
@@ -659,6 +659,18 @@ module.exports = (User = (function () {
       return 'expired'
     }
 
+    getSeenPromotion (key) {
+      const seenPromotions = this.get('seenPromotions') || {}
+      return seenPromotions[key]
+    }
+
+    setSeenPromotion (key) {
+      const seenPromotions = this.get('seenPromotions') || {}
+      Object.assign(seenPromotions, { [key]: true })
+      this.set('seenPromotions', seenPromotions)
+      this.save()
```

Comment:
I'm not a fan of this way of calling save, I would much rather let caller do the save since they might be setting multiple properties and it can lead to just 1 save by combining all the SETs and then saving. If all functions did save like this, it would lead to multiple saves being called.